File: src/main/java/example/ModTest.java
package example;

public class ModTest {

    public static int intVar1,intVar2;
    public static float realVar;

    public static void main(String[] args) {
        intVar1 = 10;
        intVar2 = 2;
        //0
        System.out.println(intVar1 % intVar2);

        //2
        System.out.println(8 % 3);

        realVar = 8 % 3;
        //2.0
        System.out.println(realVar);
    }

}

File: src/test/resources/driver/testPascalRun/modTest.pas
(*
 * Project: pascal_test
 * User: Lenovo
 * Date: 13/02/2022
 *)
program modTest(output);
var intVar1,intVar2:Integer;
realVar:Real;
begin
  intVar1:=10;
  intVar2:=2;
  WriteLn(intVar1 mod intVar2); {0}

  WriteLn(8 mod 3);{2}

  realVar:= 8 mod 3;
  WriteLn(realVar);{ 2.0000000000000000E+000}

end.